src/System.Private.ServiceModel/src/System/ServiceModel/Channels/SslStreamSecurityUpgradeProvider.cs

```diff
@@ -262,14 +262,21 @@ private void SetupServerCertificate(SecurityToken token)
                 throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(
                     SR.InvalidTokenProvided, _serverTokenProvider.GetType(), typeof(X509SecurityToken))));
             }
+
+#if TARGETS_WINDOWS
             _serverCertificate = new X509Certificate2(x509Token.Certificate.Handle);
```

Comment:
As far as I can tell, the _serverCertificate is never used for clients so this is effectively dead code. We only call SetupServerCertificate when _serverTokenProvider is not null. The only place where _serverTokenProvider is set to anything except null is in the second constructor method on line 46 which is never used. I would rather remove dead code than add more conditional statements to dead code :smile: